go-controller/pkg/ovn/pods.go

```diff
@@ -261,6 +263,26 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 
+	// the IPs we allocate in this function need to be released back to the
+	// IPAM pool if there is some error in any step of addLogicalPort past
+	// the point the IPs were assigned via the IPAM manager.
+	// this needs to be done only when releaseIPs is set to true (the case where
+	// we truly have assigned podIPs in this call) AND when there is no error in
+	// the rest of the functionality of addLogicalPort. The success variable is either set to nil
+	// in case of errors or set to a valid truth value in case everything works well.
+	// the second OR conditional is redundant in some ways, but it makes the check more explicit.
+
+	defer func(nodeName string) {
```

Comment:
you don't need this function parameter, you can directly use `logicalSwitch` in `oc.lsManager.ReleaseIPs(logicalSwitch)` instead of the current `oc.lsManager.ReleaseIPs(nodeName)`

Reply:
Agreed.